src/base/TypeUtil.ml

```diff
@@ -443,6 +443,28 @@ module TypeUtilities = struct
   let rec map_depth mt =
     match mt with MapType (_, vt) -> 1 + map_depth vt | _ -> 0
 
+  let address_field_type f t =
+    match t with
+    | Address fts ->
+        if String.(get_id f = ContractUtil.balance_label)
+        then
+          pure ContractUtil.balance_typ
+        else
+          let loc_removed = List.map fts ~f:(fun (f, t) -> (get_id f, t)) in
+          (match List.Assoc.find loc_removed (get_id f)
+                   ~equal:String.( = ) with
+          | Some ft ->
+              pure ft
+          | None ->
+              (* May be the _balance field, which is accessible but cannot be declared *)
```

Comment:
I don't understand this comment. You're already explicitly handling `_balance` above.

Reply:
I forgot to remove the comment when I added the check for the `_balance` field. I have removed it now.